astropy/io/fits/util.py

```diff
@@ -882,3 +882,23 @@ def _free_space_check(hdulist, dirname=None):
             hdu._close()
 
         raise OSError(error_message + str(exc))
+
+
+def _extract_number(value, default):
+    """
+    Attempts to extract an integer number from the given value. If the
+    extraction fails, the value of the 'default' argument is returned.
+    """
+
+    try:
```

Comment:
I'm not sure but this certainly shares a lot of code with the `_is_int` function, right?